Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm

```diff
@@ -623,11 +622,34 @@
 {
     auto operation = adoptNS([[WKQLThumbnailLoadOperation alloc] initWithURL:filePath identifier:identifier]);
     requestThumbnailWithOperation(operation.get());
-    
 }
 
 #endif // HAVE(QUICKLOOK_THUMBNAILING)
 
+#if PLATFORM(MAC)
+
+void WebPageProxy::updateIconForDirectory(NSFileWrapper* fileWrapper, const String& identifier)
+{
+    auto image = [fileWrapper icon];
+    if (!image)
+        return;
+
+    auto flippedIcon = [NSImage imageWithSize:IntSize(400, 400) flipped:YES drawingHandler:^BOOL(NSRect dstRect) {
```

Comment:
Nit - could we pull out the `IntSize(400, 400)` into a named `constexpr`?

(Ideally, `dstRect` would be `destinationRect` as well, to match WebKit style)